### ISSUE TITLE
DeviceClient on initial disconnect reports incorrect connection status

### ISSUE THREAD

**Issue by abhipsaMisra** [Azure/azure-iot-sdk-csharp#1646]
If a device client is in a disconnected state immediately after initialization, it reports a {`Disconnected`, `Retry_Expired`} event, even though it is reconnecting internally.
It should report a `Disconnected_Retrying` status instead.

**Comment by abhipsaMisra** [comment]
This has now been released: https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2020-11-5

**Comment by az-iot-builder-01** [comment]
@abhipsaMisra, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)

**Comment by generateui** [comment]
I'd like to add some information from our environment on this issue. It solved a problem, so this may help the dev team and other users.

1.	Around 2 months ago, we noticed a drop in connected gateways on an int env (non-prd) IoT Hub
2.	After closer inspection, we noticed a few (around 20 out of 40) IoT Hub devices having trouble reconnecting on int env
3.	Some of these devices were persistent in their problematic reconnection behavior
4.	A few weeks later, we also noticed one device exhibiting the exact same behavior on prd
5.	Upon inspecting the log of that prd IoTHub device, we noticed that it failed to reconnect
6.	A script was in place to reboot the IoT device every 24h to circumvent the problem
7.	We deployed a version of our software based on the SDK version with this fix
8.	We observed a correct working IoT device succeeding to reconnect
